### PR TITLE
Sentinel Deployment Fix

### PR DESCRIPTION
**Pull request by Springstone**
## Overview/Summary

This pull request contains changes to the `eslzArm/eslzArm.json` and `eslzArm/subscriptionTemplates/logAnalyticsWorkspace.json` files. The changes mainly revolve around the removal of the `enableSecuritySolution` parameter and related deployment references. Additionally, there is an update to the API version in the `logAnalyticsWorkspace.json` file.

Removal of `enableSecuritySolution`:

* [`eslzArm/eslzArm.json`](diffhunk://#diff-58a2a55a8192e9f53dbbf24829de525fd2dfe86dd56d9ea5fe4ce54f71eac970L206-L213): Removed the `enableSecuritySolution` parameter and its allowed values.
* [`eslzArm/eslzArm.json`](diffhunk://#diff-58a2a55a8192e9f53dbbf24829de525fd2dfe86dd56d9ea5fe4ce54f71eac970L1599): Removed the `monitoringSolutions` deployment URI.
* [`eslzArm/eslzArm.json`](diffhunk://#diff-58a2a55a8192e9f53dbbf24829de525fd2dfe86dd56d9ea5fe4ce54f71eac970L1717): Removed the `monitoringSolutionsDeploymentName` variable.
* [`eslzArm/eslzArm.json`](diffhunk://#diff-58a2a55a8192e9f53dbbf24829de525fd2dfe86dd56d9ea5fe4ce54f71eac970L1845): Removed the `monitoringSolutionsLiteDeploymentName` variable.
* [`eslzArm/eslzArm.json`](diffhunk://#diff-58a2a55a8192e9f53dbbf24829de525fd2dfe86dd56d9ea5fe4ce54f71eac970L2541-L2574): Removed the deployment block for Sentinel to Log Analytics workspace.
* [`eslzArm/eslzArm.json`](diffhunk://#diff-58a2a55a8192e9f53dbbf24829de525fd2dfe86dd56d9ea5fe4ce54f71eac970L7584-L7620): Removed the deployment block for Sentinel to the Log Analytics workspace for the ES Lite.

Removal of `logAnalyticsSolutions.json`:

* [`eslzArm/subscriptionTemplates/logAnalyticsSolutions.json`](diffhunk://#diff-6307b39f2f58affe4898e2f0928ac0cca3979f7ac2a632a27034a4f722ee23b2L1-L330): The entire file has been removed.

API version update:

* [`eslzArm/subscriptionTemplates/logAnalyticsWorkspace.json`](diffhunk://#diff-5440aa1959616175e19ebd6a10910d48753f2c4b4c18f19aeb86b89fe662baeeL37-R37): Updated the API version for the deployment.

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#view/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fspringstone%2FEnterprise-Scale%2FSentinel%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fspringstone%2FEnterprise-Scale%2FSentinel%2FeslzArm%2Feslz-portal.json)

### CODE REVIEW

**Comment by jtracey93** [comment]
@arjenhuitema can you confirm none of the solutions for LAW are needed anymore?

**Comment by Springstone** [comment]
> @arjenhuitema can you confirm none of the solutions for LAW are needed anymore?

None are currently used. Everything was commented out except the old way of deploying Sentinel.

**Comment by arjenhuitema** [comment]
> @arjenhuitema can you confirm none of the solutions for LAW are needed anymore?

Correct, there is no need to deploy any of the legacy solutions anymore. Sentinel was the last one.

Note that the ChangeTracking solution is still installed but this is done automatically as part of the CT DCR deployment.

File: eslzArm/subscriptionTemplates/logAnalyticsWorkspace.json [hunk 1]
```diff
@@ -85,6 +85,19 @@
...
+                  "apiVersion": "2023-02-01-preview",
+                  "type": "Microsoft.SecurityInsights/onboardingStates",
+                  "name": "default",
+                  "scope": "[concat(subscription().id, '/resourceGroups/', parameters('rgName'), '/providers/Microsoft.OperationalInsights/workspaces/', parameters('workspaceName'))]",
+                  "dependsOn": [
+                      "[concat(subscription().id, '/resourceGroups/', parameters('rgName'), '/providers/Microsoft.OperationalInsights/workspaces/', parameters('workspaceName'))]"
+                  ],
+                  "properties": {
+                      "customerManagedKey": false
+                  }
```
Comment (jtracey93) [inline eslzArm/subscriptionTemplates/logAnalyticsWorkspace.json:100]: No condition on this????

**Review by jtracey93 — CHANGES_REQUESTED** [review]

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**